icefoganalytics/travel-authorization
# Show All Basic User Information On Users Profile Page

Fixes https://github.com/icefoganalytics/travel-authorization/issues/88

# Context

See https://github.com/icefoganalytics/travel-authorization/issues/55

1. Add all basic fields to the users profile page:
    - email
    - status
    - first_name
    - last_name
    - department
    - roles
    - division
    - branch
    - unit
    - mailcode
    - manager

2. User information should sync for a given user the first time they log in that day.
3. User profile page should have a force-sync button that will refresh the users profile information from the government directory at time of click.

# Implementation

Add User role translations to web/src/locales/en.js.
Replace "profile" store with "current-user/index.ts" store. I should probably flatten this to `current-user.ts` but I can't quite figure out the namespacing in JS to avoid putting data in the index file.

# Testing Instructions

1. Boot the app via `dev up`.
2. Log in to the app at http://localhost:8080/
3. Go to the profile page via the "kebab" menu in the top right, then selecting, "My profile".
4. Check that the information that shows up is correct.
5. Check that you can force a sync with the external government directory.
